### PR TITLE
Jet Energy Correction uncertainty groupings

### PR DESCRIPTION
**Pull request by dwolfschlaeger**
This PR implements the possibility to define and add groupings of JEC uncertainties. In Artus a macro was added to read in the dictionary which assigns a list of individual uncertainties to a groupname. The TaggedJetUncertaintyProducer is now capable of adding these grouping names to the list of uncertainty names and calcules the resulting jet shifts by summing the member uncertainties up quadratically. 
This way a more elegant way to calculate also the "Closure" uncertainty was achieved (It is now defined as a grouping as well). Local tests prove the Producer to work as expected.

### CODE REVIEW

**Comment by jd550179** [comment]
I will look at this beginning next week

File: scripts/makePlots_datacardsCPInitialState.py [hunk 1]
```diff
@@ -686,9 +693,10 @@ def remove_procs_and_systs_with_zero_yield(proc):
 	
 	if args.n_plots[0] != 0 and "t2w" in args.steps:
 		# tools.parallelize(_call_command, hadd_commands, n_processes=args.n_processes)
-		datacards_module._call_command([
-			"$CMSSW_BASE/src/HiggsAnalysis/KITHiggsToTauTau/scripts/hadd_shapes.sh {OUTPUT_FOLDER}".format(OUTPUT_FOLDER=os.path.join(args.output_dir, "shapes", args.output_suffix))
-		])
+		for channel in ["em","et","mt","tt","ttbar"]:
```
Comment (pfackeldey) [inline scripts/makePlots_datacardsCPInitialState.py]: ttbar as a channel is misleading. Could you explain why you implement it like this?

File: scripts/makePlots_datacardsCPInitialState.py [hunk 2]
```diff
@@ -686,9 +693,10 @@ def remove_procs_and_systs_with_zero_yield(proc):
...
+			datacards_module._call_command([
+				"$CMSSW_BASE/src/HiggsAnalysis/KITHiggsToTauTau/scripts/hadd_shapes.sh {OUTPUT_FOLDER} {CHANNEL} {CHN}".format(OUTPUT_FOLDER=os.path.join(args.output_dir, "shapes", args.output_suffix), CHANNEL=channel if channel != "ttbar" else "em", CHN=channel if channel != "ttbar" else "em")
```
Comment (pfackeldey) [inline scripts/makePlots_datacardsCPInitialState.py]: difference CHANNEL and CHN? again see review comment above

File: scripts/makePlots_datacardsCPInitialState.py [hunk 3]
```diff
@@ -653,7 +658,8 @@ def remove_procs_and_systs_with_zero_yield(proc):
 					) for sample in config["labels"]]
 					tmp_output_file = os.path.join(args.output_dir, tmp_input_root_filename_template.replace("$", "").format(
 							ANALYSIS="htt",
-							CHANNEL=channel,
+							CHANNEL=chn,
+							CHN=chn,
```
Comment (pfackeldey) [inline scripts/makePlots_datacardsCPInitialState.py]: see review comment above...

File: .gitignore [hunk 4]
```diff
@@ -12,3 +12,4 @@ wiki
 param.log
 latex
 html
+.vs*
```
Comment (pfackeldey) [inline .gitignore]: why do you need to add this to the gitignore? I don't see a new feature in your PR which adds these kind of files.

**Pull request merged**